server/repository/src/db_diesel/temperature_chart/temperature_chart.rs

```diff
@@ -53,20 +59,34 @@ impl<'a> TemperatureChartRepository<'a> {
             .load::<QueryResult>(&self.connection.connection)?
             .into_iter()
             .map(TemperatureChartRow::from)
-            .collect();
+            .collect::<Result<_, _>>()?;
 
         Ok((chart_data, intervals))
     }
 }
 
-impl From<QueryResult> for TemperatureChartRow {
-    fn from((from_datetime, to_datetime, average_temperature, sensor_id): QueryResult) -> Self {
-        Self {
+impl TemperatureChartRow {
+    fn from(
+        (from_datetime, to_datetime, average_temperature, sensor_id, breach_ids): QueryResult,
+    ) -> Result<Self, RepositoryError> {
+        // breach_ids at this stage is a stringified JSON array of string or null values
+        // i.e. '[null, "first_breach"]
+        let breach_ids_with_nulls: Vec<Option<String>> = serde_json::from_str(&breach_ids)
+            .map_err(|e| RepositoryError::DBError {
+                msg: e.to_string(),
+                extra: breach_ids,
+            })?;
+
+        Ok(Self {
             from_datetime,
             to_datetime,
             average_temperature,
             sensor_id,
-        }
+            breach_ids: breach_ids_with_nulls
+                .into_iter()
+                .filter_map(|id_or_none| id_or_none)
```

Comment:
filter_map would filter out none values and keeps the Some values (unwrapped), and at this stage null values are `None`